services/grpc/grpc.js

```diff
@@ -205,6 +207,45 @@ class GrpcService extends EventEmitter {
       }
     }
   }
+
+  /**
+   * Wait for a service to become active.
+   * @return {Promise<Object>} Object with `isActive` and `cancel` properties.
+   */
+  /**
+   * Wait for a service to become active.
+   * @param  {String} serviceName Name of service to wait for (Lightning, or WalletUnlocker)
+   * @return {Promise<Object>}     Object with `isActive` and `cancel` properties.
+   */
+  waitForService(serviceName) {
+    let successHandler
+    const activationEventName = `GRPC_${snakecase(serviceName).toUpperCase()}_SERVICE_ACTIVE`
+
+    /**
+     * Promise that resolves when service is active.
+     */
+    const isActive = new Promise(async resolve => {
+      // If the service is already active, return immediatly.
+      if (this.services[serviceName].is('connected')) {
+        return resolve()
+      }
+      // Otherwise, wait until we receive an activation event from the gRPC service.
+      successHandler = () => resolve()
```

Comment:
To me it looks like if `isActive` is called multiple times `successHandler` will get overridden and event listener won't get cleared. Should we clear previous listener here?

Reply:
`IsActive` is not a method. It is a promise. If the user does `await isActive` multiple times, they are waiting on the same promise, so no new listeners are being setup.

Reply:
Right, my bad, was looking at `cancel` and automatically implied `isActive` is also a method